pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -1248,9 +1266,11 @@ public ResultSet getProcedureColumns(@Nullable String catalog, @Nullable String
     f[17] = new Field("ORDINAL_POSITION", Oid.INT4);
     f[18] = new Field("IS_NULLABLE", Oid.VARCHAR);
     f[19] = new Field("SPECIFIC_NAME", Oid.VARCHAR);
-
-    String sql;
-    sql = "SELECT current_database(), n.nspname,p.proname,p.prorettype,p.proargtypes, t.typtype,t.typrelid, "
+    // if the catalog is specified and it does not equal the current catalog then just return an empty resultset
+    if (catalog != null && !catalog.isEmpty() && !catalog.equals(connection.getCatalog())) {
```

Comment:
`"" retrieves those without a catalog`
I assume PostgreSQL does not have "procedures without a catalog", so `catalog.isEmpty()` check looks redundant here. WDYT of replacing this with `catalog != null && (catalog.isEmpty() || !catalog.equals(connection.getCatalog()))`

`!Objects.equals(catalog, connection.getCatalog())` could do as well (at a cost of one-time SQL in `getCatalog()`)